compiler/IREmitter/NameBasedIntrinsics.cc

```diff
@@ -732,6 +732,45 @@ class InstanceVariableGet : public NameBasedIntrinsicMethod {
     }
 } InstanceVariableGet;
 
+class InstanceVariableSet : public NameBasedIntrinsicMethod {
+public:
+    InstanceVariableSet() : NameBasedIntrinsicMethod{Intrinsics::HandleBlock::Unhandled} {};
+
+    virtual llvm::Value *makeCall(MethodCallContext &mcctx) const override {
+        if (mcctx.send->args.size() != 2) {
+            return IREmitterHelpers::emitMethodCallViaRubyVM(mcctx);
+        }
+
+        auto &cs = mcctx.cs;
+        auto &builder = mcctx.builder;
+        auto &var = mcctx.send->args[0].type;
+        if (!core::isa_type<core::LiteralType>(var)) {
+            return IREmitterHelpers::emitMethodCallViaRubyVM(mcctx);
+        }
+        auto lit = core::cast_type_nonnull<core::LiteralType>(var);
+        if (lit.literalKind != core::LiteralType::LiteralTypeKind::Symbol) {
+            return IREmitterHelpers::emitMethodCallViaRubyVM(mcctx);
+        }
```

Comment:
What about supporting string literals as well? It looks like it's converted to a symbol automatically: https://ruby-doc.org/core-2.7.2/Object.html#method-i-instance_variable_set

Looking into `rb_obj_ivar_set` it seems like this might not be super straight forward, but it would be nice to handle in the future.

Reply:
I think it would be straightforward to handle in the future, but as strings don't come up for cases this patch is intended to accelerate, I'm comfortable leaving it this way for now.